contracts/extensions/contracts/MilestonesExtension.sol

```diff
@@ -0,0 +1,233 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+// Interfaces
+import {IMilestonesExtension} from "../interfaces/IMilestonesExtension.sol";
+// Core Contracts
+import {CoreBaseStrategy} from "../../strategies/CoreBaseStrategy.sol";
+// Internal Libraries
+import {Metadata} from "../../core/libraries/Metadata.sol";
+import {Errors} from "../../core/libraries/Errors.sol";
```

Comment:
maybe we can remove this and just add a custom error which is more descriptive instead of `UNAUTHORIZED` ?